types/src/write_set.rs

```diff
@@ -144,6 +144,40 @@ impl WriteOp {
     }
 }
 
+pub enum WriteOpSize {
```

Comment:
Just a thought to consider having struct WriteOpSize with named deposit: Option<u64>, len: u64 (or Option<u64>), and kind: Creation/Deletion/DeletionWithDeposit?

I feel like the focus here can be on size / deposit as opposed to the op itself, e.g. given that the enum is called WriteOpSize and some variants have (u64), it seems an easy mistake for the reader to assume the u64 would be the size? Moreover, it seems like we could have deposit method alongside the write_len method and avoid straight up matches (I've commented on a few above, but those comments apply nonetheless)

Reply:
as mentioned above comments cannot be addressed.

for here, I'll change to have name fields for enum options here - but otherwise type is generally what is matched on - and we have utility functions to get the fields you want

Reply:
I'll use write_len as field name, as that is the name of the function - if you have preference for something else (i.e. value_size) let me know

Reply:
I don't think it is cleaner to to have kind enum, and Options for other fields, that depend on the state of the enum.

I.e. kind=delete, size=Some would be invalid state, or kind=create, deposit=Some

with enum having fields that only refer to it, it is guaranteed to be internally consistent.

and so places that only care about one dimension (i.e. size), can get Option of it. cases that care about distinction will be able to match - and not need to deal with matching on create and then calling unwrap() on size, for example.